lib/cn/csched_sp3_work.c

```diff
@@ -323,6 +323,65 @@ sp3_work_leaf_garbage(
     return min_t(uint, kvsets, thresh->lcomp_kvsets_max);
 }
 
+static uint
+sp3_work_leaf_scatter(
+    struct sp3_node *         spn,
+    struct sp3_thresholds *   thresh,
+    struct kvset_list_entry **mark,
+    enum cn_action *          action,
+    enum cn_comp_rule *       rule)
+{
+    struct kvset_list_entry *le;
+    struct cn_tree_node *tn;
+    struct list_head *head;
+    uint kvsets;
+
+    tn = spn2tn(spn);
+
+    head = &tn->tn_kvset_list;
+    *mark = list_last_entry(head, typeof(*le), le_link);
+    *action = CN_ACTION_COMPACT_KV;
+    *rule = CN_CR_LSCATF;
+
+    kvsets = cn_ns_kvsets(&tn->tn_ns);
+
+    /* Find the oldest kvset which has vgroup scatter...
+     */
+    list_for_each_entry_reverse(le, head, le_link) {
+        if (kvset_get_vgroups(le->le_kvset) > 1) {
+            uint compc = kvset_get_compc(le->le_kvset);
+
+            *mark = le;
+
+            /* Include older kvsets which have the same compc
+             * (to prevent compc misordering).
+             */
+            while (( le = list_next_entry_or_null(le, le_link, head) )) {
+                if (kvset_get_compc(le->le_kvset) > compc)
```

Comment:
Is compc maintained even for kv-compacts?

Reply:
I"m pretty certain it is always incremented for both k and kv compactions.  I've often pondered having kv-compact not do that, but it would change things quite a bit..